File: snippets/java/search_batch_points.java
import java.util.List;

import static io.qdrant.client.ConditionFactory.matchKeyword;

import io.qdrant.client.QdrantClient;
import io.qdrant.client.QdrantGrpcClient;

import io.qdrant.client.grpc.Points.Filter;
import io.qdrant.client.grpc.Points.SearchPoints;

QdrantClient client =
    new QdrantClient(QdrantGrpcClient.newBuilder("localhost", 6334, false).build());

Filter filter = Filter.newBuilder().addMust(matchKeyword("city", "London")).build();
List<SearchPoints> searches =
    List.of(
        SearchPoints.newBuilder()
            .addAllVector(List.of(0.2f, 0.1f, 0.9f, 0.7f))
            .setFilter(filter)
            .setLimit(3)
            .build(),
        SearchPoints.newBuilder()
            .addAllVector(List.of(0.5f, 0.3f, 0.2f, 0.3f))
            .setFilter(filter)
            .setLimit(3)
            .build());
client.searchBatchAsync("{collection_name}", searches, null).get();

File: snippets/csharp/search_batch_points.cs
using Qdrant.Client;
using Qdrant.Client.Grpc;
using static Qdrant.Client.Grpc.Conditions;

var client = new QdrantClient("localhost", 6334);

var filter = MatchKeyword("city", "London");

var searches = new List<SearchPoints>
{
  new()
  {
    Vector = { new float[] { 0.2f, 0.1f, 0.9f, 0.7f } },
    Filter = filter,
    Limit = 3
  },
  new()
  {
    Vector = { new float[] { 0.5f, 0.3f, 0.2f, 0.3f } },
    Filter = filter,
    Limit = 3
  }
};

await client.SearchBatchAsync(collectionName: "{collection_name}", searches: searches);
